### ISSUE TITLE
Minibuffer's hide function can reference wrong minibuffer/window pair

### ISSUE THREAD

**Issue by tviti** [atlas-engineer/next#471]
**Describe the bug**

If a new window is created and activated before `read-from-minibuffer`
completes, then the subsequent call to `(rpc-window-active)` in `(hide
minibuffer)` will end up returning the new window instead of the old one (which
makes it impossible for the original window to process any further keypress
events, since `minibuffer` doesn't actually exist in that window's
`active-minibuffer` list).

Changing the `let` block in `hide` to use `(last-active-window *interface*)`
instead of `(rpc-window-active)` seems to work around this, but doesn't properly
clear the echo area.

**Precise recipe to reproduce the issue**

From `next-pyqt`, open the minibuffer w/ `M-x`, and then create a new window
with `make-window<RETURN>`

With `hide` modified to:

```	
 (defun hide (minibuffer)
   "Hide MINIBUFFER and display next active one, if any."
+  (log:info "active-window: ~s" (id (last-active-window *interface*)))
   (let ((active-window (rpc-window-active)))
+    (log:info "active-window: ~s" (id active-window))
     ;; Note that MINIBUFFER is not necessarily first in the list, e.g. a new
     ;; minibuffer was invoked before the old one reaches here.
     (setf (active-minibuffers active-window)
...
```

The recipe yields the output:
```
 <INFO> [14:49:00] next minibuffer.lisp (hide) - active-window: "1"
 <INFO> [14:49:01] next minibuffer.lisp (hide) - active-window: "2"
```

Despite window 2 not actually having a minibuffer at this stage.

**Information**
- OS name+version: macOS 10.14.6
- Kernel name+version: darwin?
- Graphics card and driver: intel iris 6100
- Desktop environment / Window manager name+version: 
- Next version (from =M-x next-version= or =next --version=): 1.3.3
- How you installed Next (Guix pack, package manager, build from source): build
  from source
- SBCL version (if built from source): 1.5.6 (from `homebrew`)
- WebKitGTK+ / PyQt-Webengine version: pyqt-webengine w/ modifications

**Comment by Ambrevar** [comment]
Indeed.

I believe to fix this properly we would need a separate echo area and
status  buffer.

This will come with window management.  But maybe sooner would be a good
idea because those "bugs" have been annoying for too long :p

**Comment by tviti** [comment]
> I believe to fix this properly we would need a separate echo area and status buffer.

Yeah, I thought I might be able to sort this out by making class `minibuffer`
hold a reference to its parent window, and then using that in a call to
`%echo-status`, but I tried, and it looks like `%echo-status` doesn't really
care what window you give it (still doesn't clear the echo/status area, even
when explicitly passed the correct window).

**Comment by Ambrevar** [comment]
Indeed, it's a semantic problem because the "echo" function are also
used to clear the minibuffer, which makes little sense.  We would need
to have to separate display areas.

**Comment by Ambrevar** [comment]
Will be fixed with #608.

**Comment by jmercouris** [comment]
I've tested, and this is now fixed on master :-)
